pghoard/postgres_command.py

```diff
@@ -46,9 +54,19 @@ def restore_command(site, xlog, output, host=PGHOARD_HOST, port=PGHOARD_PORT):
         output_path = os.path.join(os.getcwd(), output)
         headers = {"x-pghoard-target-path": output_path}
         method = "GET"
-    conn = HTTPConnection(host=host, port=port)
-    conn.request(method, "/{}/archive/{}".format(site, xlog), headers=headers)
-    status = conn.getresponse().status
+    path = "/{}/archive/{}".format(site, xlog)
+
+    for retries in range(retry_count - 1, -1, -1):
+        try:
+            status = http_request(host, port, method, path, headers)
+            break
+        except socket.error as ex:
```

Comment:
consider also https://docs.python.org/2/library/httplib.html#httplib.HTTPException or some of its subclasses? Some of the sub-classes like IncompleteRead such applicable here.